### PR TITLE
Introduce markdownlint #864

### PR DESCRIPTION
**Pull request by roottool**
## Description

I introduced [markdownlint](https://github.com/DavidAnson/markdownlint) in Boostnote.
markdownlint

### Note
This feature checks [all rules and aliases](https://github.com/DavidAnson/markdownlint#rules--aliases).
I didn't implement ignore settings on purpose.
Because I think that it will exceed the range dealt with in #864.

## Issue fixed

#864 

## Type of changes

- :white_circle: Bug fix (Change that fixed an issue)
- :white_circle: Breaking change (Change that can cause existing functionality to change)
- :white_circle: Improvement (Change that improves the code. Maybe performance or development improvement)
- :radio_button: Feature (Change that adds new functionality)
- :white_circle: Documentation change (Change that modifies documentation. Maybe typo fixes)

## Checklist:

- :radio_button: My code follows [the project code style](docs/code_style.md)
- :white_circle: I have written test for my code and it has been tested
- :radio_button: All existing tests have been passed
- :radio_button: I have attached a screenshot/video to visualize my change if possible

### CODE REVIEW

File: browser/components/CodeEditor.js [hunk 1]
```diff
@@ -344,7 +377,12 @@ export default class CodeEditor extends React.Component {
       inputStyle: 'textarea',
       dragDrop: false,
       foldGutter: true,
-      gutters: ['CodeMirror-linenumbers', 'CodeMirror-foldgutter'],
+      lint: {
+        'getAnnotations': validatorOfMarkdown,
+        'async': true
+      },
+      mode: 'markdown',
```
Comment (ZeroX-DG) [inline browser/components/CodeEditor.js]: Please don't use `markdown` mode here. We set the mode according to the props or self-detecting language result if the CodeEditor is used in snippet note.
https://github.com/BoostIO/Boostnote/blob/885f656d3458af090c62f81b4f8a1c775effcdc6/browser/components/CodeEditor.js#L357-L361

Comment (roottool) [reply browser/components/CodeEditor.js]: I understood.
I fixed it. Please check it again.

**Comment by roottool** [comment]
We need writing `config.json` or create a new settings form in interface settings(red circle of below image) to set [markdownlint settings](https://github.com/DavidAnson/markdownlint#optionsconfig).  
boostnotesettings

But probably I can't do it for a while.
Because I will be busy from next week for my private.

And if Implementing markdownlint settings includes this PR, I think it is difficult to test because the coverage is too wide.
After resolving the implementation of markdownlint, I think it is a good idea to create a new issue.

**Comment by Rokt33r** [comment]
@roottool Could you resolve the conflicts?

**Comment by roottool** [comment]
@Rokt33r Sure! I resolved the conflicts.
Please check the code again.

**Comment by kleinjm** [comment]
How do you disable this?

**Comment by judsoniv** [comment]
It would be great to get those preferences. I find the it distracting to see the errors for things I do not care about for my purposes, such as using hard tabs or line length. And it seems to be on by default and not able to be disabled? I don't think it would have expanded the scope too much to just have a stub on-off toggle in preferences until the full lint-preferences section was ready.

**Comment by roottool** [comment]
I'm working on it to be able to edit the markdownlint rule at #2983.
markdownlint is turn off by [json](https://github.com/DavidAnson/markdownlint#optionsconfig).
Therefore, I had to add a new config editor in preferences of Boostnote.

**Comment by niorko** [comment]
This feature bothers me a lot, how can I disable it? :) Thanks

**Comment by Zorlin** [comment]
In the same boat... did not expect this change and it's really annoying with seemingly no easy way to disable it.

**Comment by takacsot** [comment]
I just got the fresh update and this feature is F^#$ing annoying. 

And there is not option to switch off!!!!

I am almost exclusively using the editor mode and those warnings are sh@#t.

**Comment by judsoniv** [comment]
**roottool:** It's not fun to get negative feedback on a feature. I want to thank you for contributing to the Boostnote product. This feature, once more mature, will be one that I will use and enjoy. I thought I would take a minute to expand on my pervious comment with my thoughts after using this for a day.

Here's what I think would make this feature prime-time for me:

- Toggle in preferences to turn the entire feature on or off
- Toggles in preferences to enable or disable the individual markdownlint rules
- Warning icon should be on the line that has the error, rather than the next line

Again, I appreciate shipping software, and I understand that you're contributing to this project because you want to, not because you have to. You try to do something nice for people, and users like me yell at you. Sorry about that.

I look forward to your next iteration.

**Comment by bkj** [comment]
Any way to disable? Or if not, is there an easy way to roll back my version?

**Comment by jeffisabelle** [comment]
Hi @roottool, 

Thanks for your work on this, I'm pretty sure hearing complaints about your opensource contributions sucks, please consider this as constructive feedback. 

It's probably nice to have a markdown linter for many cases but for my use case, it somewhat breaks all the fun I have with boostnote. I tend to store a lot of logs under a code snippet, now it throws a bunch of linter exceptions. I actually don't care about the markdown linting on this case. It'd be great if we can have a toggle in preferences to disable linting. 

Also, I noticed linting makes the switch between editor mode and preview mode a lot slower. 

For people who have similar problems, I rolled back to the previous version manually by installing it from here;
https://github.com/BoostIO/boost-releases/releases/download/v0.11.15/Boostnote-mac.zip

**Comment by Zorlin** [comment]
It looks like the latest update to Boostnote has removed or hidden this feature.

I would like to take the time to say thank you for building this, even if I didn't like it - as an optional feature I think it's great.

**Pull request merged**